iOSChatExample/iOSChatExample/Models/ChatManager.swift

```diff
@@ -0,0 +1,203 @@
+                onRecievedMessage: { message in
+                    self.handleIncomingMessage(message)
+                }
+            )
+        }
+    }
+    
+    // Handle incoming message
+    // Update messages and send appropriate events
+    func handleIncomingMessage(_ message: Message) {
+        
+        // To remove typing indicator from previous messages
+        self.messages.removeAll { $0.text == "..." }
+        // Check for message receipts
+        if (message.contentType == ContentType.metaData.rawValue) {
+            // To update the message receipts
+            if let row = self.messages.firstIndex(where: {$0.messageID == message.messageID && !$0.text.isEmpty}) {
+                self.messages[row].status = message.status
+            }
+            // Do not show typing event for customer
+        }else if !(message.text == "..." && message.participant == config.customerName){
+            // Show the message as usual
+            DispatchQueue.main.async {
+                self.messages.append(message)
+            }
+        }
+        
+        // Sends a `Delivered` event back to agent.
+        if (!message.text.isEmpty && message.participant == config.agentName && message.contentType.contains("text")){
+            let content = "{\"messageId\":\"\(message.messageID!)\"}"
+            self.sendEvent(contentType: .messageDelivered, content: content)
+        }
+    }
+    
+    func sendReadEventOnAppear(
+        message: Message
+    ){
+        if let row = self.messages.firstIndex(where: {
+            $0.text == message.text && !$0.text.isEmpty && $0.contentType.contains(
+                "text"
+            ) && $0.participant != config.customerName &&
+            !$0.isRead
+        }) {
+            let content = "{\"messageId\":\"\(messages[row].messageID!)\"}"
+            sendEvent(
+                contentType: .messageRead,
+                content: content
+            )
+            messages[row].isRead = true
+        }
+    }
+    
+    
+    // MARK: - API and SDK calls
+    
+    /// Initiates a flow to start a new chat for the customer. Response of this API provides a token required to obtain credentials from the CreateParticipantConnection API in the Amazon Connect Participant Service.
+    /// - Parameters:
+    ///  - name: Information identifying the participant as part of  ParticipantDetails Object.
+    ///  - Attributes  : A custom key-value pair using an attribute map
+    ///  - ContactFlowId  : The identifier of the flow for initiating the chat. To see the ContactFlowId in the Amazon Connect console user interface, on the navigation menu go to Routing, Contact Flows.     Choose the flow. On the flow page, under the name of the flow, choose Show additional flow information. The ContactFlowId is the last part of the ARN.
+    ///  - InstanceId  : The identifier of the Amazon Connect instance. You can find the instance ID in the Amazon Resource Name (ARN) of the instance.
```

Comment:
suggestion: could we also link to [Admin guide docs](https://docs.aws.amazon.com/connect/latest/APIReference/API_StartChatContact.html) or the IOS SDK docs